hedera-mirror-rest/service/tokenService.js

```diff
@@ -18,20 +18,71 @@
  * ‍
  */
 
-import _ from 'lodash';
-import {Token} from '../model';
+import {TokenAccount, TokenBalance, AccountBalanceFile, ContractResult} from '../model';
+import BaseService from './baseService';
+import {OrderSpec} from '../sql';
 
 /**
  * Token retrieval business logic
  */
-class TokenService {
-  static tokenByIdQuery = `select *
-                           from ${Token.tableName}
-                           where ${Token.TOKEN_ID} = $1`;
-
-  async getToken(tokenId) {
-    const {rows} = await pool.queryQuietly(TokenService.tokenByIdQuery, tokenId);
-    return _.isEmpty(rows) ? null : new Token(rows[0]);
+class TokenService extends BaseService {
+  static accountBalanceQuery = `select max(${AccountBalanceFile.CONSENSUS_TIMESTAMP})
+                from ${AccountBalanceFile.tableName}`;
+
+  static tokenBalanceJoin = `left join (select ${TokenBalance.TOKEN_ID},${TokenBalance.BALANCE}
+                from ${TokenBalance.tableName}
+                where ${TokenBalance.ACCOUNT_ID} = $1
+                and ${TokenBalance.CONSENSUS_TIMESTAMP} = (${TokenService.accountBalanceQuery})) ${
+    TokenBalance.tableAlias
+  }
+                on ${TokenAccount.getFullName(TokenAccount.TOKEN_ID)} = ${TokenBalance.getFullName(
+    TokenBalance.TOKEN_ID
+  )}`;
+
+  static tokenRelationshipsQuery = `select ${TokenAccount.getFullName(TokenAccount.AUTOMATIC_ASSOCIATION)},
+            ${TokenAccount.getFullName(TokenAccount.CREATED_TIMESTAMP)},
+            ${TokenAccount.getFullName(TokenAccount.FREEZE_STATUS)},
+            ${TokenAccount.getFullName(TokenAccount.KYC_STATUS)},
+            ${TokenAccount.getFullName(TokenAccount.TOKEN_ID)},
+       ${TokenBalance.getFullName(TokenBalance.BALANCE)} from ${TokenAccount.tableName} ${TokenAccount.tableAlias}
+       ${TokenService.tokenBalanceJoin}
+       where ${TokenAccount.tableAlias}.${TokenAccount.ACCOUNT_ID} = $1`;
+
+  /**
+   * Gets the full sql query and params to retrieve an account's token relationships
+   *
+   * @param query
+   * @return {{sqlQuery: string, params: *[]}}
+   */
+  getQuery(query) {
```

Comment:
Since this class will eventually be used for more than just relationships

```suggestion
  getTokenRelationshipsQuery(query) {
```